### ISSUE TITLE
`fill_between()` doesn't work with DatetimeIndex

### ISSUE THREAD

**Issue by oguzhanogreden** [lukelbd/proplot#225]
### Description

When DatetimeIndex is used for x axis in fill_between, the wrapper doesn't work.

### Steps to reproduce

```python
import proplot as plot
import pandas as pd

df = pd.DataFrame(1, index=pd.date_range('2020-09-01', '2020-09-02', freq='1H'), columns=['test'])
_, axes = plot.subplots()

axes.plot(df)

axes.fill_between(
    df.index,
    df['test'] + 1,
    df['test'] + 1.1,
)

# Raises `TypeError: '<' not supported between instances of 'Timestamp' and 'float'`
```

I expect this to be possible, it's quite a common use case for my use.

### Equivalent steps in matplotlib

```python

import matplotlib as mpl
import matplotlib.pyplot as plt

import pandas as pd

mpl.__version__ # 3.3.1

df = pd.DataFrame(1, index=pd.date_range('2020-09-01', '2020-09-02', freq='1H'), columns=['test'])

axes = plt.subplot()

axes.plot(df)

axes.fill_between(
    df.index,
    df['test'] + 1,
    df['test'] + 1.1,
)

```

This works.

### Proplot version

'0.6.4.post28'

I realise this isn't the latest, but I see that this file hasn't changed since this version. Let me know if you think an update can solve the issue and I'll give it a try.

Thanks! I really appreciate the comfort of using `proplot` - can't wait for the stable version! :)

**Comment by lukelbd** [comment]
Thanks! This is now fixed. This was a regression introduced by #166, which makes the bottom and sides of area plots "sticky", so there is no gap between the subplot edge and the shading bounds by default (similar to `bar`). Matplotlib's approach is kind of opaque, and I failed to account for non-numeric data when implementing it.

**Comment by lukelbd** [comment]
Closed by: 7a8e9652]

**Comment by oguzhanogreden** [comment]
Thanks a lot!
